src/analyzer/plugins/analyzerqueenmarybeats.cpp

```diff
@@ -9,8 +9,9 @@
 
 namespace mixxx {
 namespace {
-
-constexpr float kStepSecs = 0.01161; // stepsize should be equivalent to ~11.61 ms irrespectively of samplerate
+// stepsize should be equivalent to ~10ms and this constant (11.61ms) make the number 
+// of samples an exact power of 2 for 22.1k, 44.1k, and very close to, for 96k and 192k
+constexpr float kStepSecs = 0.01161; 
```

Comment:
The sample rate is passed to this analyzer in `initialize`. How about making this a private member variable and setting it in `initialize` to the best value for the sample rate?

Reply:
This makes a lot of sense actually, and going this route we may completely get rid of this constant, we can just multiply the sample rate times 0.01 and round the result to the closest power of two to get the best step size

Reply:
It's generally good practice to pull out constants that aren't obvious and properly document them, then the code below that uses them is easier to understand and verify because it refers to a constant name instead of a literal. 

As for rounding to the closest power of 2, that would potentially speed up the FFT but I'd rather we stick to how the VAMP plugin previously worked and then treat that as a follow-up improvement if it's not quality-netural.

Reply:
The reasoning behind this constant is not documented anywhere in original plugin code other than "512 samples for 44100hz"
And after putting a lot of thought on it, I am pretty confident the only reason is indeed that. With some research, I found a [paper](https://www.eecs.qmul.ac.uk/~josh/documents/2010/Zhou%20Reiss%20-%20Music%20Onset%20Detection%202010.pdf) from QM C4M, that mention the 10ms stepsize.
So I think we should move forward with this and get rid of the weird constant and make the stepsize a nice power of 2 not only for 44100 and 22100 but for every sample rate.
And with this, we also get rid of that weird undocumented addition of 0.0001  in `(m_iSampleRate * kStepSecs + 0.0001)`

Reply:
@rryan so merge this now and leave the refactoring for a follow up PR?

Reply:
> And after putting a lot of thought on it, I am pretty confident the only reason is indeed that.

Right -- people often use powers of two for window sizes so that you can take an FFT without Bluestein's algorithm. As a designer you usually have a desired frequency and time resolution you're shooting for, and you pick the closest power-of-2 that achieves that for the sample rate you're using for efficiency. `window_size = 2 *step_size` is also a common rule of thumb if you are doing synthesis so that the window function you are using has the [COLA](https://ccrma.stanford.edu/~jos/sasp/Constant_Overlap_Add_COLA_Cases.html) property. This allows perfect reconstruction if you are doing analysis -> synthesis.

But this is about the design of the algorithm. Once you've picked a desired time resolution and frequency resolution, you don't want that to change because the parameters of the model have been tuned based on that.

What you're saying is that we are going to change the time and frequency resolution of the algorithm based on the sampling rate of the audio file. Ideally the algorithm would produce identical results for the same song when at 24 kHz or 44.1 kHz -- this change would make that less likely.

Reply:
Nice, thank you for the technical explanation. But I am not sure if I get everything.
> 
> What you're saying is that we are going to change the time and frequency resolution of the algorithm based on the sampling rate of the audio file. Ideally the algorithm would produce identical results for the same song when at 24 kHz or 44.1 kHz -- this change would make that less likely.

Isn't that what we are doing with ` m_stepSize = (m_iSampleRate * 0.01161 + 0.0001)`

What I am suggesting is to change it `m_stepSize = pow(2, ceil(log(m_iSampleRate * 0.01)/log(2)))`

It would generate the same stepsize values for 22.1khz or 44.1 kHz, which are 256 and 512 respectively. Which I believe was exactly the reason they use this constant instead of 10ms.
But then we would also generate power of 2s for other sample rates as well.

>
> But this is about the design of the algorithm. Once you've picked a desired time resolution and frequency resolution, you don't want that to change because the parameters of the model have been tuned based on that.

That 512 samples for 44100 comment is probably a strong indication that the algorithm was tuned for this only and this would still hold true

Reply:
NVM the above, I think I got you now.
The time resolution should be fixed at 11.61ms because that's how the algorithm was tuned to work. 
That's it, right?